File: src/features/compliance/dto/strip-pii.dto.ts
import { IsIn, IsString } from 'class-validator';
import { COMPLIANCE_ENTITY_TYPES, ComplianceEntityType } from '../compliance.types';

export class StripPiiDto {
  @IsString()
  @IsIn([...COMPLIANCE_ENTITY_TYPES])
  entityType: ComplianceEntityType;

  @IsString()
  entityId: string;
}

File: spring-boot-api/src/main/java/com/mytegroup/api/dto/compliance/StripPiiDto.java
package com.mytegroup.api.dto.compliance;

import jakarta.validation.constraints.NotBlank;

public record StripPiiDto(
    @NotBlank(message = "Entity type is required")
    String entityType,
    @NotBlank(message = "Entity ID is required")
    String entityId
) {
    public StripPiiDto {
        if (entityType != null) {
            entityType = entityType.trim();
        }
        if (entityId != null) {
            entityId = entityId.trim();
        }
    }
}

